libavcodec/vaapi_decode.c

```diff
@@ -584,7 +570,8 @@ static int vaapi_decode_make_config(AVCodecContext *avctx,
 }
 
 int ff_vaapi_common_frame_params(AVCodecContext *avctx,
```

Comment:
How about add new function, ff_vaapi_common_frame_params_with_pdb_size. then ff_vaapi_common_frame_params will call ff_vaapi_common_frame_params_with_pdb_size with size dpb_size = 2.
In this way. you do not need change any code in mpeg2, vc1.... etc.